### PR TITLE
feat: logs & events

### PR DESCRIPTION
**Pull request by qiweiii**
- [x] api
- [x] ui

tests will be added after change to new designs

https://github.com/Magickbase/godwoken_explorer/issues/494

### CODE REVIEW

File: pages/tx/[hash].tsx [hunk 1]
```diff
@@ -53,16 +54,19 @@ import {
...
 } from 'utils'
 
-const tabs = ['erc20']
 type ParsedTransferList = ReturnType<typeof getERC20TransferListRes>
 type RawTx = Parameters<typeof getTxRes>[0]
 type ParsedTx = ReturnType<typeof getTxRes>
+type State = ParsedTx & Partial<{ transferList: ParsedTransferList; logsList: ParsedEventLog[] }>
 
+const tabs = ['erc20', 'logs']
 type State = ParsedTx & Partial<{ transferList: ParsedTransferList }>
```
Comment (Keith-CY) [inline pages/tx/[hash].tsx:70]:
```suggestion
```

**Comment by Keith-CY** [comment]
How is it going

**Comment by qiweiii** [comment]
Contract Events
image

Tx Logs
image

File: utils/api/logs.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  txHash: string
+  data: string
+  blockNumber: number
+  topics: string[]
+}
+
+export const getEventLogsListRes = (raw: Raw): ParsedEventLog[] => {
+  return raw.data.map(({ attributes, id }) => {
+    const log3 = [attributes.first_topic, attributes.second_topic, attributes.third_topic]
+    const log4 = [...log3, attributes.fourth_topic]
```
Comment (Keith-CY) [inline utils/api/logs.ts:70]: It's not the correct way to handle `null` topic since topics[0...3] could be `null` too.
Use `0x` if a topic is a `null`

**Review by Keith-CY — APPROVED** [review]

**Pull request merged**